integration-tests/src/test/scala/chiselTest/util/experimental/TapSpec.scala

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
This test is looking great. One other idea is to test at every level. I.e., make sure that this thing definitely works. You could do something like:

```scala
object TapAndCheck {
  def apply(tapSignal: UInt) = {
    val readTap = Wire(UInt(8.W))
    readTap := tapAndRead(tapSignal)
    chisel3.assert(readTap === 123.U)
  }
}
```

Then use that in every module to check that it works, including in the same module as the tap. It would then also be good to check one or two levels down, i.e., to add two levels below leftlower and rightlower. This is clearly overkill.